3_javascript/javascript1/week2/script2.js

```diff
@@ -0,0 +1,181 @@
+console.log("Full Name 2:", fullname2);
+
+
+
+  // Formal fullname Part //
+  function getFullname(firstname, surname, useFormalName, isFemale) {
+    if (useFormalName) { return "Lord " + firstname + surname; }
+    else if (!firstname && !surname) { return "Please provide a valid name."; }
+    else if (isFemale) { return "Lady " + firstname + surname; }
+    else { return firstname + surname; }
+  }
+  
+  const formalFullName = getFullname("Milad ", "Fetrat", true, false);
+  const inFormalFullName = getFullname("Amin ", "Hussaini", false, false);
+  const formalFullFemaleName = getFullname("Hanna ", "Jensen", false, true);
+  const missingNames = getFullname("", "", false);
+  
+  console.log("Formal Full Name:", formalFullName);
+  console.log("Informal Full Name:", inFormalFullName);
+  console.log("Formal Full Female Name:", formalFullFemaleName);
+  console.log("Missing Names:", missingNames);
+  
+
+
+  //Event application//
+
+  function getEventWeekday(daysFromToday) {
+    const weekdays = ['Sunday', 'Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday'];
+    const today = new Date();
+    const eventDate = new Date(today);
+    eventDate.setDate(today.getDate() + daysFromToday);
+    const eventWeekdayIndex = eventDate.getDay();
+    const eventWeekday = weekdays[eventWeekdayIndex];
+   return eventWeekday;
+  };
+  
+const today = new Date().toLocaleDateString('en-US', { weekday: 'long' });
+console.log(`Today is ${today}`);
+
+console.log(getEventWeekday(7)); // Logs out "Thursday" (if today is Tuesday)
+console.log(getEventWeekday(2)); // Logs out "Sunday" (if today is Friday)
+
+
+
+
+
+  //Weather wear function//
+
+
+  function suggestClothing(temperature) {
+    if (temperature >= 30) {
+      return "You should wear shorts and a tank top.";
+    } else if (temperature >= 20 && temperature < 30) {
+      return "You can wear jeans and a light sweater.";
+    } else if (temperature >= 10 && temperature < 20) {
+      return "It's a bit chilly. Wear long pants and a jacket.";
+    } else {
+      return "It's quite cold. Bundle up with a heavy coat and scarf.";
+    }
+  }
+  
+  const clothesToWear = suggestClothing(5);
+  console.log(clothesToWear); 
+
+
+
+
+  //Student manager//
+  const class07Students = [];
+
+  function addStudentToClass(studentName) {
+    if (studentName === "") {
+      console.log("You must add a name!");
+    } else if (class07Students.includes(studentName)) {
+      console.log("This student is already in the class.");
+    } else {
+      if (studentName.toLowerCase() === "queen") {
+        class07Students.push(studentName);
+        console.log("Queen has been added to the class!");
+      } else if (class07Students.length >= 6) {
+        console.log("Cannot add more students to class 07.");
+      } else {
+        class07Students.push(studentName);
+        console.log(`The student ${studentName} has been added to the class.`);
+      }
+    }
+  }
```

Comment:
The code here could be simplified and the conditions reordered. A tip to do this is to check the general conditions first and then go to the particular conditions.

Besides this, the code should work. Well done.